transmit-lib/src/cmr/transmit/tokens.clj

```diff
@@ -17,14 +18,17 @@
    [cmr.transmit.connection :as conn]
    [cmr.transmit.launchpad-user-cache :as launchpad-user-cache]))
 
-(defn verify-edl-token-locally
-  "Uses the EDL public key to verify jwt tokens locally."
+(defn verify-json-web-token-locally
+  "Uses a known public JWKS to verify JWT tokens locally."
   [token]
   (try
-    (let [public-key (buddy-keys/jwk->public-key (json/parse-string (transmit-config/edl-public-key) true))
-          bearer-stripped-token (string/replace token #"Bearer\W+" "")
+    (let [bearer-stripped-token (string/replace token #"Bearer\W+" "")
+          token-kid (get (jws/decode-header bearer-stripped-token) :kid)
+          jwks-list (json/parse-string (transmit-config/jwt-web-key-set) true)
+          matching-key (first (filter #(= token-kid (get % :kid)) jwks-list))
+          public-key (buddy-keys/jwk->public-key matching-key)
           decrypted-token (jwt/unsign bearer-stripped-token public-key {:alg :rs256})]
-      (:uid decrypted-token))
+      (:uid decrypted-token (:username decrypted-token)))
```

Comment:
This line should probably move to using a configuration-based strategy.  We should define
```clojure
(defconfig jwt-user-id-claim {:default "uid"})
```
and then fetch the user ID as
```clojure
((keyword jwt-user-id-claim) decrypted-token)
```